Add documentation for custom resources, fix other docs as well (#112)
kubeshop/kusk-gateway: docs/customresources/index.md

> # Custom Resouces

Reviewer: ```suggestion
# Custom Resources
```

> Custom Resources is the Kubernetes concept to extend K8s API with third party APIs.

Reviewer 2: ```suggestion
Custom Resource Definition is the Kubernetes concept to extend K8s API with third party APIs.
```

> They're deployed as YAML manifests files and are picked up by Kusk Gateway Manager to configure Envoy settings and routing.

Reviewer: ```suggestion
They're deployed as YAML manifests files and are picked up by the Kusk Gateway Manager to configure Envoy settings and routing.
```
